make/mt-config/mt-config

```diff
@@ -56,7 +56,12 @@ ifneq (,$(findstring debug,$(TARGET)))
 	XI_DEBUG_EXTRA_INFO ?= 1
 endif
 
-XI_COMPILER_FLAGS += -Wall -Werror
+# Settings that will work only on linux and only against clang-4.0 and greater
+ifneq (,$(findstring fuzz_test,$(CONFIG)))
+    XI_CONFIG_FLAGS += -fsanitize=address -fomit-frame-pointer -fsanitize-coverage=trace-pc-guard -g
+endif
+
+XI_COMPILER_FLAGS += -Wall
```

Comment:
I am not certaint of -Wall, there are some compilers where some were intentionally turned off e.g. mt-arm-linux: XI_COMPILER_FLAGS += -Wno-format. Another reason for holding back this -Wall is the false positives of Travis. So here we cannot rely on Travis' greens.